src/sentry/static/sentry/app/views/alerts/rules/details/metricChart.tsx

```diff
@@ -41,6 +63,47 @@ function createThresholdSeries(lineColor: string, threshold: number): LineChartS
   };
 }
 
+function createStatusAreaSeries(
```

Comment:
thanks for pulling these things into functions 🙏 

One thing to maybe note is that you might not require a whole new series for each individual status area in the graph. I think you can just supply additional arrays of [{coord: <>, {coord: <>}] in the `data:` field of the markLine. This way you would only need three series for the whole chart, a green series (`resolved`), yellow (`warning`), and red (`critical`).

Not sure if that would help the chart render faster, but could be something to keep in mind

Reply:
It would be negligible compared to time it takes to fetch the events-stats data